src/intelli/gui/ui/node/boolnodeui.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * GTlab IntelliGraph
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *  SPDX-FileCopyrightText: 2025 German Aerospace Center
+ *
+ *  Author: Marius Bröcker <marius.broecker@dlr.de>
+ */
+
+#include <intelli/gui/ui/node/boolnodeui.h>
+#include <intelli/data/bool.h>
+#include <intelli/node/booldisplay.h>
+#include <intelli/node/input/boolinput.h>
+
+#include <intelli/gui/graphics/nodeobject.h>
+#include <intelli/gui/widgets/booldisplaygraphicswidget.h>
+
+using namespace intelli;
+
+BoolNodeUI::BoolNodeUI() = default;
+
+NodeUI::WidgetFactoryFunction
+BoolNodeUI::centralWidgetFactory(Node const& n) const
```

Comment:
I really like the refactoring. This separates the computational part from the ui part and should (in theory) also work better in batch contexts. I like, that the node types to not need to know their ui-counterparts anymore. Good job!